### PR TITLE
checkpoint management console connector

### PR DESCRIPTION
**Pull request by cs-omkar-kumbhar**
### Descriptions:
PMDB: 27948

#### UTCs:
- [x] Connector installation verified.
- [x] Connector logo verified.
- [x] Actions and Playbook list verified.
- [x] Playbook tags verified.
- [x] All Playbook are in info mode verified.
- [x] All Playbook are in inactive mode verified.

### CODE REVIEW

File: checkpoint-management-console/operations.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            if "/login" not in endpoint:
+                headers["X-chkp-sid"] = self.token
+            headers["Content-Type"] = "application/json"
+
+            logger.debug(f"\n---------------req-start---------------\n{method}-{endpoint}\nheaders: {headers}\nparams: {params}\ndata: {data}")
+            try:
+                from connectors.debug_utils.curl_script import make_curl
+                make_curl(method, endpoint, headers=headers, params=params, data=data, verify_ssl=self.verify_ssl)
+            except Exception as err:
+                logger.error(f"Error in curl utils: {str(err)}")
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/operations.py]: This logger will generate unnecessary logs if customer is not using Debug Utils connector. Remove this log line or change logger level

File: checkpoint-management-console/info.json [hunk 2]
```diff
@@ -0,0 +1,789 @@
...
+          "title": "Name",
+          "required": true,
+          "visible": true,
+          "editable": true,
+          "type": "text",
+          "name": "name",
+          "tooltip": "Specify the host name. Must be unique in the domain.",
+          "description": "Specify the host name. Must be unique in the domain."
+        },
+        {
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/info.json:102]: We can provide a dropdown to select which type of IP address to add , and make it required. As IP Address, IPv4 Address and IPv6 Address are optional could result in playbook failure if not provided atleast one of them

File: checkpoint-management-console/info.json [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+              "subnet-mask": "255.255.255.0",
+              "color": "blue",
+              "comments": "any comment",
+              "details-level": "standard",
+              "ignore-warnings": false,
+              "ignore-errors": false
+            }
+          }
+        },
+        {
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/info.json:371]: Same change mentioned above should be done here for IP address.

File: checkpoint-management-console/info.json [hunk 4]
```diff
@@ -0,0 +1,789 @@
...
+            "UID",
+            "Standard",
+            "Full"
+          ],
+          "name": "details-level",
+          "tooltip": "Specify the details level. The level of detail for some of the fields in the response can vary from showing only the UID value of the object to a fully detailed representation of the object.",
+          "description": "Specify the details level. The level of detail for some of the fields in the response can vary from showing only the UID value of the object to a fully detailed representation of the object.",
+          "value": "standard"
+        },
+        {
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/info.json]: Can be added onchange parameter of Details Level. As it does not supported for Full option in Details Level.

File: checkpoint-management-console/info.json [hunk 5]
```diff
@@ -0,0 +1,789 @@
...
+    {
+      "operation": "get_host_details",
+      "description": "Retrieve host details based on the uid or name that you have specified.",
+      "category": "investigation",
+      "annotation": "get_host_details",
+      "title": "Get Host Details",
+      "enabled": true,
+      "parameters": [
+        {
+          "title": "UID or Name",
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/info.json]: Title can be changed to `Based On`.

File: checkpoint-management-console/playbooks/playbooks.json [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+              "top": "120",
+              "stepType": "/api/3/workflow_step_types/0bfed618-0316-11e7-93ae-92361f002671"
+            }
+          ],
+          "triggerLimit": null,
+          "description": "Update host based on the parameters that you have specified.",
+          "name": "Update Host",
+          "tag": "#CheckPoint Management Console",
+          "recordTags": [
+            "Checkpoint",
```
Comment (cs-sameer-pathan) [inline checkpoint-management-console/playbooks/playbooks.json]: P should be in capital letters . `CheckPoint`

**Review by cs-sameer-pathan — CHANGES_REQUESTED** [review]

File: checkpoint-management-console/operations.py (hunk 1 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/operations.py]: done

File: checkpoint-management-console/info.json (hunk 2 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/info.json:102]: there are many usecases of it and we dont have setup to verify, so it is better to keep it like this

File: checkpoint-management-console/info.json (hunk 4 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/info.json]: onchange parameter is not required here

File: checkpoint-management-console/info.json (hunk 5 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/info.json]: "UID or Name" is more meaningful then "Based On"

File: checkpoint-management-console/playbooks/playbooks.json (hunk 6 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/playbooks/playbooks.json]: that is autogenerated tag so it should be fine

File: checkpoint-management-console/info.json (hunk 5 above)
Comment (cs-sameer-pathan) [reply checkpoint-management-console/info.json]: In other connectors we have provided Based On parameter for similar use cases. Suggested to make it consistent

File: checkpoint-management-console/info.json (hunk 4 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/info.json]: done

File: checkpoint-management-console/info.json (hunk 5 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/info.json]: done

File: checkpoint-management-console/playbooks/playbooks.json (hunk 6 above)
Comment (cs-omkar-kumbhar) [reply checkpoint-management-console/playbooks/playbooks.json]: done

**Review by cs-sameer-pathan — APPROVED** [review]
LGTM

**Review by cs-prerna-joshi — APPROVED** [review]

**Pull request merged**